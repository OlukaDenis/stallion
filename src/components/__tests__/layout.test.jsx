import React from 'react';
import render, { initialState } from '../../test_utils';
import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import BaseLayoutConnected, { BaseLayout } from '../layout';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let store;
let baseLayoutComponent;

test('BaseLayout Component: main layout component should match snapshot', () => {
  store = mockStore(initialState);

  baseLayoutComponent = renderer.create(
    <Provider store={store}>
      <BaseLayoutConnected t={(key) => key}></BaseLayoutConnected>
    </Provider>
  );
  expect(baseLayoutComponent.toJSON()).toMatchSnapshot();
});

test('BaseLayout Component: Renders the connected BaseLayout with initialState', () => {
  store = mockStore(initialState);
  baseLayoutComponent = render(<BaseLayoutConnected></BaseLayoutConnected>, { initialState: initialState });
  expect(screen.getAllByText(/button.theme.light/i)).toHaveLength(2);
});
