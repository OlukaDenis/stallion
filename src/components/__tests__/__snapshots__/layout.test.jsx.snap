// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout Component: main layout component should match snapshot 1`] = `
<section
  className="ant-layout fill-screen"
>
  <header
    className="ant-layout-header site-header site-header-light"
    onClick={[Function]}
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <picture>
        <source
          srcSet="/images/webp/logo.webp"
          type="image/webp"
        />
        <img
          alt="Super Stallion Logistics Logo"
          className="site-header-logo"
          src="/images/logo.jpg"
        />
      </picture>
    </a>
    <button
      aria-checked={true}
      className="ant-switch theme-mode-switch ant-switch-checked"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      >
        button.theme.light
      </span>
    </button>
    <div
      className="vertical-center-div trigger-light"
    >
      <span
        className="menu-badge-icon badge-icon ant-badge"
      >
        <span
          aria-label="menu-fold"
          className="anticon anticon-menu-fold"
          onClick={[Function]}
          role="img"
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu-fold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
            />
          </svg>
        </span>
      </span>
    </div>
  </header>
  <section
    className="ant-layout ant-layout-has-sider site-layout"
  >
    <main
      className="ant-layout-content no-margin"
      onClick={[Function]}
    >
      <div
        className="ant-spin min-full-height"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </main>
    <aside
      className="floating-side-menu ant-layout-sider ant-layout-sider-light ant-layout-sider-collapsed"
      style={
        Object {
          "flex": "0 0 80px",
          "maxWidth": "80px",
          "minWidth": "80px",
          "width": "80px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <picture>
          <source
            srcSet="/images/webp/logo.webp"
            type="image/webp"
          />
          <img
            alt="Super Stallion Logistics Logo"
            className="menu-header-logo"
            src="/images/logo.jpg"
          />
        </picture>
        <div
          className="theme-mode-switch-in-sider"
        >
          <button
            aria-checked={true}
            className="ant-switch ant-switch-checked"
            data-testid="ThemeModeSwitchInSideMenu"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              button.theme.light
            </span>
          </button>
        </div>
        <div
          className="small-margin-h-divider ant-divider ant-divider-horizontal"
          role="separator"
        />
        <ul
          className="ant-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          direction="ltr"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-submenu ant-menu-submenu-vertical"
            data-testid="LanguageChooserButton"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-menu-submenu-title"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="button"
              style={Object {}}
            >
              <span
                aria-label="edit"
                className="anticon anticon-edit"
                role="img"
              >
                <picture>
                  <source
                    srcSet="/images/webp/icon128px-exported-black.webp"
                    type="image/webp"
                  />
                  <img
                    alt="Language Icon"
                    className="custom-icon-image"
                    src="/images/icon128px-exported-black.jpg"
                  />
                </picture>
              </span>
              <span>
                menu.language.label
              </span>
              <i
                className="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
        <div
          className="no-margin ant-divider ant-divider-horizontal"
          role="separator"
        />
        <ul
          className="ant-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          direction="ltr"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="home"
              className="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span>
              menu.home.label
            </span>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="cloud-server"
              className="anticon anticon-cloud-server"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="cloud-server"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M704 446H320c-4.4 0-8 3.6-8 8v402c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8V454c0-4.4-3.6-8-8-8zm-328 64h272v117H376V510zm272 290H376V683h272v117z"
                />
                <path
                  d="M424 748a32 32 0 1064 0 32 32 0 10-64 0zm0-178a32 32 0 1064 0 32 32 0 10-64 0z"
                />
                <path
                  d="M811.4 368.9C765.6 248 648.9 162 512.2 162S258.8 247.9 213 368.8C126.9 391.5 63.5 470.2 64 563.6 64.6 668 145.6 752.9 247.6 762c4.7.4 8.7-3.3 8.7-8v-60.4c0-4-3-7.4-7-7.9-27-3.4-52.5-15.2-72.1-34.5-24-23.5-37.2-55.1-37.2-88.6 0-28 9.1-54.4 26.2-76.4 16.7-21.4 40.2-36.9 66.1-43.7l37.9-10 13.9-36.7c8.6-22.8 20.6-44.2 35.7-63.5 14.9-19.2 32.6-36 52.4-50 41.1-28.9 89.5-44.2 140-44.2s98.9 15.3 140 44.3c19.9 14 37.5 30.8 52.4 50 15.1 19.3 27.1 40.7 35.7 63.5l13.8 36.6 37.8 10c54.2 14.4 92.1 63.7 92.1 120 0 33.6-13.2 65.1-37.2 88.6-19.5 19.2-44.9 31.1-71.9 34.5-4 .5-6.9 3.9-6.9 7.9V754c0 4.7 4.1 8.4 8.8 8 101.7-9.2 182.5-94 183.2-198.2.6-93.4-62.7-172.1-148.6-194.9z"
                />
              </svg>
            </span>
            <span>
              menu.track.label
            </span>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="unordered-list"
              className="anticon anticon-unordered-list"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="unordered-list"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <span>
              menu.quotation.label
            </span>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="unlock"
              className="anticon anticon-unlock"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="unlock"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 464H332V240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v68c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-68c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zm-40 376H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                />
              </svg>
            </span>
            <span>
              menu.login.label
            </span>
          </li>
        </ul>
        <div
          className="small-margin-h-divider ant-divider ant-divider-horizontal"
          role="separator"
        />
        <ul
          className="ant-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          direction="ltr"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="info"
              className="anticon anticon-info"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="info"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M448 224a64 64 0 10128 0 64 64 0 10-128 0zm96 168h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8h64c4.4 0 8-3.6 8-8V400c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span>
              menu.about.label
            </span>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="phone"
              className="anticon anticon-phone"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="phone"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 01-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 00-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 00285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
                />
              </svg>
            </span>
            <span>
              menu.contact.label
            </span>
          </li>
        </ul>
        <div
          className="small-margin-h-divider ant-divider ant-divider-horizontal"
          role="separator"
        />
        <ul
          className="ant-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          direction="ltr"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="eye-invisible"
              className="anticon anticon-eye-invisible"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="eye-invisible"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                />
                <path
                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                />
              </svg>
            </span>
            <span>
              menu.privacy.label
            </span>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="solution"
              className="anticon anticon-solution"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="solution"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M688 264c0-4.4-3.6-8-8-8H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48zm-8 136H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM480 544H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm-48 308H208V148h560v344c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V108c0-17.7-14.3-32-32-32H168c-17.7 0-32 14.3-32 32v784c0 17.7 14.3 32 32 32h264c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm356.8-74.4c29-26.3 47.2-64.3 47.2-106.6 0-79.5-64.5-144-144-144s-144 64.5-144 144c0 42.3 18.2 80.3 47.2 106.6-57 32.5-96.2 92.7-99.2 162.1-.2 4.5 3.5 8.3 8 8.3h48.1c4.2 0 7.7-3.3 8-7.6C564 871.2 621.7 816 692 816s128 55.2 131.9 124.4c.2 4.2 3.7 7.6 8 7.6H880c4.6 0 8.2-3.8 8-8.3-2.9-69.5-42.2-129.6-99.2-162.1zM692 591c44.2 0 80 35.8 80 80s-35.8 80-80 80-80-35.8-80-80 35.8-80 80-80z"
                />
              </svg>
            </span>
            <span>
              menu.terms.label
            </span>
          </li>
        </ul>
        <div
          className="small-margin-h-divider ant-divider ant-divider-horizontal"
          role="separator"
        />
        <ul
          className="ant-menu bottom-padded-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
          direction="ltr"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
          >
            <span
              aria-label="caret-left"
              className="anticon anticon-caret-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="caret-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M689 165.1L308.2 493.5c-10.9 9.4-10.9 27.5 0 37L689 858.9c14.2 12.2 35 1.2 35-18.5V183.6c0-19.7-20.8-30.7-35-18.5z"
                />
              </svg>
            </span>
            <span>
              menu.show.label
            </span>
          </li>
        </ul>
      </div>
    </aside>
  </section>
  <div
    className="ant-back-top"
    onClick={[Function]}
  />
  <footer
    className="ant-layout-footer site-footer site-footer-light"
    onClick={[Function]}
  >
    <div
      className="vertical-center-div"
    >
      <p>
        Super Stallion Logistics ©
        2020
         | Designed by:
         
        <a
          href="https://premar.tech"
        >
          Premar Systems
        </a>
      </p>
    </div>
  </footer>
</section>
`;
